libobs/graphics/device-exports.h

```diff
@@ -177,6 +177,15 @@ EXPORT void device_debug_marker_begin(gs_device_t *device,
 				      const char *markername,
 				      const float color[4]);
 EXPORT void device_debug_marker_end(gs_device_t *device);
+EXPORT bool device_is_monitor_hdr(gs_device_t *device, void *monitor);
+EXPORT bool device_shared_texture_available(void);
+
+#ifdef __APPLE__
+EXPORT gs_texture_t *device_texture_create_from_iosurface(gs_device_t *device,
+							  void *iosurf);
+EXPORT gs_texture_t *device_texture_open_shared(gs_device_t *device,
+						uint32_t handle);
+#endif
```

Comment:
Is it possible to PR these and the other `libobs/graphics` changes separately?

Reply:
Some yes, others not.

Switching `glad` to a static library (as done by the rework) requires code changes to make it work:

`glad` usually needs to be generated by an online API and the headers are generated according to the "flavour" of the library needed. Thus `#define GLAD_GLAPI_EXPORT` needs to be removed for it to work as a static library.

Moving `GL_SILENCE_DEPRECATION` into the compiler definitions is also directly tied to the CMake change.

Reply:
Have glad files being re-generated with the right toolling ?
Because manual modification is not good for auto-generated files.

Nitpik: The glad generator can be run locally.

Reply:
It's a manual change because I have no idea which version we used to generate the variant in our sources and the last time I tried to update it I was in a world of pain.

`GLAD_GLAPI_EXPORT` only enables the `dllimport`/`dllexport` declarations for Windows and the visibility preset for clang/gcc in the `GLAPI` keyword, neither of which are needed for static linking.

Reply:
So I went back to the glad v1 sources it looks like this define was set as a compiler option depending on whether a shared build was desired or not:

https://github.com/Dav1dde/glad/blob/1ecd45775d96f35170458e6b148eb0708967e402/CMakeLists.txt#L157

Looking at the glad v2 generator, it also only adds the define to enable shared library support and nothing else.